src/parse.rs

```diff
@@ -307,7 +302,15 @@ impl<'a> Parser<'a> {
                 None
             };
             if let Some(group) = group {
-                return Ok((ix + skip, Expr::Backref(group)));
+                self.backrefs.insert(group);
```

Comment:
I wasn't sure about this part initially (and `&self` -> `&mut self`) but thinking about alternatives, it seems best to modify backrefs here, at the place where we create the corresponding Expr, not in a higher place (depending on what we returned before, as the old code did).